plugins/storage/volume/scaleio/src/main/java/org/apache/cloudstack/storage/datastore/manager/ScaleIOSDCManagerImpl.java

```diff
@@ -51,9 +53,17 @@
 import com.cloud.utils.exception.CloudRuntimeException;
 
 @Component
-public class ScaleIOSDCManagerImpl implements ScaleIOSDCManager {
+public class ScaleIOSDCManagerImpl implements ScaleIOSDCManager, Configurable {
     private Logger logger = LogManager.getLogger(getClass());
 
+    static ConfigKey<Boolean> ConnectOnDemand = new ConfigKey<>("Storage",
+            Boolean.class,
+            "powerflex.connect.on.demand",
+            Boolean.FALSE.toString(),
+            "Connect PowerFlex client on Host when first Volume created and disconnect when last Volume deleted (or always stay connected otherwise).",
```

Comment:
```suggestion
            "Connect PowerFlex client on Host when first Volume is mapped to SDC and disconnect when last Volume is unmapped from SDC," +
                    " otherwise no action (that is connection remains in the same state whichever it is, connected or disconnected).",
```